LongWarOfTheChosen/Src/LW_PerkPack_Integrated/Classes/X2Ability_XMBPerkAbilitySet.uc

```diff
@@ -2925,6 +2939,72 @@ static function X2AbilityTemplate Magnum()
 	return Passive('Magnum_LW', "img:///UILibrary_XPerkIconPack.UIPerk_pistol_sniper", false, Effect);
 }
 
+
+static function X2AbilityTemplate CrusaderRage()
+{
+	local XMBEffect_ConditionalBonus Effect;
+	local X2Condition_UnitStatCheck Condition;
+	local X2Effect_GreaterPadding GreaterPaddingEffect;
+	local X2AbilityTemplate Template;
+	// Create a condition that checks that the unit is at less than 100% HP.
+	// X2Condition_UnitStatCheck can also check absolute values rather than percentages, by
+	// using "false" instead of "true" for the last argument.
+	Condition = new class'X2Condition_UnitStatCheck';
+	Condition.AddCheckStat(eStat_HP, 51, eCheck_LessThan,,, true);
+
+	// Create a conditional bonus effect
+	Effect = new class'XMBEffect_ConditionalBonus';
+
+	//Need to add for all of them because apparently if you crit you don't hit lol
+	Effect.AddPercentDamageModifier(50, eHit_Success);
+	Effect.AddPercentDamageModifier(50, eHit_Graze);
+	Effect.AddPercentDamageModifier(50, eHit_Crit);
+	Effect.EffectName = 'CrusaderRage_Bonus';
+
+	// The effect only applies while wounded
+	EFfect.AbilityShooterConditions.AddItem(Condition);
+	Effect.AbilityTargetConditionsAsTarget.AddItem(Condition);
+
+	GreaterPaddingEffect = new class 'X2Effect_GreaterPadding';
+	GreaterPaddingEffect.BuildPersistentEffect (1, true, false);
+	GreaterPaddingEffect.Padding_HealHP = default.CRUSADER_WOUND_HP_REDUCTTION;	
+	
+	// Create the template using a helper function
+	Template = Passive('CrusaderRage_LW', "img:///UILibrary_XPerkIconPack.UIPerk_melee_adrenaline", true, Effect);
+	Template.AddTargetEffect(GreaterPaddingEffect);
+	return Template;
+}
+
+static function X2AbilityTemplate KnifeJugglerTrigger()
+{
+	local X2AbilityTemplate 			Template;
+	local X2Effect_AddAmmo 				AmmoEffect;
+	local X2Condition_UnitProperty		UnitPropertyCondition;
+	local X2Condition_PrimaryWeapon PrimaryWeaponCondition;
+
+	AmmoEffect = new class'X2Effect_AddAmmo';
+	AmmoEffect.ExtraAmmoAmount = 1;
+
+	
+	Template = SelfTargetTrigger('KnifeJugglerTrigger_LW', "img:///'BstarsPerkPack_Icons.UIPerk_ScrapMetal'", false, AmmoEffect, 'KillMail');
+	    
+
+	PrimaryWeaponCondition = new class'X2Condition_PrimaryWeapon';
+	PrimaryWeaponCondition.RequirePrimary = true;
```

Comment:
This condition doesn't seem to be used

Reply:
Also, why is this ability defined here rather than alongside KnifeJuggler in `X2Ability_ThrowingKnifeAbilitySet`?

Reply:
Because X2Ability_ThrowingKnifeAbilitySet does not extend XMB classes. Should I make it so it does and then move it there?

Reply:
Yes. Better to keep related things together. Changing the super class of `X2Ability_ThrowingKnifeAbilitySet` isn't a big deal.